### ISSUE TITLE
Wrong suggestion when comparing strings

### ISSUE THREAD

**Issue by frewsxcv** [Manishearth/rust-clippy#386]
```rust
#![feature(plugin)]

#![plugin(clippy)]


fn some_function() -> bool {
    let some_string = String::from("a");
    let some_str = "b";
    some_string == some_str.to_owned()
}
```

results in

```
src/lib.rs:9:20: 9:39 warning: this creates an owned instance just for comparison. Consider using `some_string.as_slice()` to compare without allocation, #[warn(cmp_owned)] on by default
src/lib.rs:9     some_string == some_str.to_owned()
                                ^~~~~~~~~~~~~~~~~~~
```

it should really suggest just dropping `to_owned()` and not suggesting `as_slice()` since `String` can deref to a `&str`
